gke-deploy/README.md

```diff
@@ -0,0 +1,205 @@
+# `gke-deploy` `Task`
+
+This Task deploys an application to a Google Kubernetes Engine cluster using [`gke-deploy`](https://github.com/GoogleCloudPlatform/cloud-builders/tree/master/gke-deploy).
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/gke-deploy/gke-deploy.yaml
+```
+
+## Inputs
+
+### Resources
+
+* **source-repo**: The Git source repository that contains your application's Kubernetes configs.
+
+### Parameters
+
+* **ARGS**: The arguments to pass to `gke-deploy` CLI.
```

Comment:
From what I recall, a Kubernetes cluster was also going to be a resource type, that basically would give you a kubeconfig file.

If so, I suspect we should support that as an optional feature (i.e. you can either pass the target cluster via a pipeline resource, or via arguments to the task)?

/cc @sbwsg on the state of Kubernetes resources

Reply:
this resource type exists but unfortunately in the current implementation resources can't be marked as optional.  this means the user would need to provide a cluster resource even if they wanted to use the args-only approach. :(

Reply:
Thanks Scott, I'll add a reminder to myself to update the pipeline with this once resources can be marked optional.

Reply:
This is the feature request in tekton for optional resources: https://github.com/tektoncd/pipeline/issues/812

Reply:
> this resource type exists but unfortunately in the current implementation resources can't be marked as optional

Ah, OK, thanks. Would you recommend leaving things as-is for now, or recommending that users create a modified the task definition for their own use if the want to use a cluster resource as input?

Reply:
Hmmm.  How would you feel about adding two tasks in this PR?  One for those users wanting to deploy using a cluster resource and one for those preferring to use args?

Reply:
Sure, I can add that in another PR, since this one has already been reviewed by docs

Reply:
sounds good.  sorry this is an admittedly lame solution. we need better support for this.

Reply:
Especially if we have a way to do this "properly" coming soon, I'd prefer to avoid creating two "official" tasks and then trying to explain to users which to choose.

For the time being, I'd rather either explain that we will add support for a cluster input when optional resources become available, or that users could use cluster inputs by making their own copy of the task, or both.

Reply:
:+1: makes sense